### PR TITLE
[stable-v2.2] cherry-pick docker and other upgrade from main branch

### PR DESCRIPTION
**Pull request by marc-hb**
95%, mostly clean cherry-picks from main branch

This avoids branching the Docker image and turns all checks green.

### CODE REVIEW

**Comment by fredoh9** [comment]
there is no sof-cnl.ri generated. apl, jsl, tgl are built fine.

```
14:09:23  [2023-05-02 21:09:23 UTC][SOF-SH] Now time: 2023-05-02T14:09:22-07:00 | Device time: 2023-05-02T21:09:23+00:00
14:09:24  [2023-05-02 21:09:24 UTC][SOF-SH] ERROR: Failed to retrieve Firmware resource from [.../pr/sof/PR7551-6887/build6887/fw/xcc/sof-cnl.ri](http://sof-storage.ostc.intel.com/pr/sof/PR7551-6887/build6887/fw/xcc/sof-cnl.ri%1B[0m)
```

**Comment by fredoh9** [comment]
For GCC build with docker image, there is an error. `docker: Error response from daemon: failed to create shim task`

https://github.com/thesofproject/sof/actions/runs/4865719524/jobs/8676405764?pr=7551

```
...
Digest: sha256:1055dadc6dd96ed5fcd3a201409772e6a52cf8b3cc2a659fcff7f121792def9a
Status: Downloaded newer image for thesofproject/sof:latest_ubuntu22.04
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "./scripts/sudo-cwd.sh": stat ./scripts/sudo-cwd.sh: no such file or directory: unknown.
time="2023-05-02T21:02:28Z" level=error msg="error waiting for container: context canceled"
Error: Process completed with exit code 127.
```

**Comment by fredoh9** [comment]
SOFCI TEST

**Comment by fredoh9** [comment]
@marc-hb I added cnl build for XCC. Need to push new patch to trigger the build/sof-test again

**Comment by marc-hb** [comment]
This is blocked by
- https://github.com/thesofproject/sof/pull/7552

\+ manual rebuild and deploy.

**Comment by fredoh9** [comment]
I built new docker image using this PR. thesofproject/sof:20230515_ubuntu22.04. I found there are some missing commits.
Oh, https://github.com/thesofproject/sof/pull/7552 is merged, then I need to re-build again with main.

**Review by cujomalainey — APPROVED** [review]

**Comment by marc-hb** [comment]
Only a couple suspend/resume failures in https://sof-ci.01.org/sofpr/PR7551/build8011/devicetest/index.html

https://sof-ci.01.org/sof-pr-viewer/#/build/PR7551/build12013250 cannot work because quickbuild CI is simply not compatible with the stable-v2.2 branch which uses a very different manifest.

@wszypelt, @aborisovich  can we please disable quickbuild on that branch? Unless someone is capable of reverting it to what it was at v2.2 time but I'm really not holding my breath.

File: .github/workflows/zephyr.yml [hunk 1]
```diff
@@ -11,15 +11,23 @@ jobs:
...
+      #   (".cached.\"WEST_TOPDIR/zephyr/arch/xtensa/include/kernel_arch_func.h\"'
+      #   [-Werror=attributes]
+
       # https://github.com/zephyrproject-rtos/docker-image
       # Note: env variables can be passed to the container with
       # -e https_proxy=...
       - name: build
         run: docker run -v "$(pwd)":/workdir
-             ghcr.io/zephyrproject-rtos/zephyr-build:latest
+             ghcr.io/zephyrproject-rtos/zephyr-build:v0.23.4
```
Comment (greg-intel) [inline .github/workflows/zephyr.yml:31]: We are explicitly changing the behavior here?  Changing from the "moving target" of "latest" to a hard-coded version?  Does that mean someone has to make sure to keep this up-to-date?

File: scripts/docker-run.sh [hunk 2]
```diff
@@ -25,12 +25,16 @@ if tty --quiet; then
...
-# Not fatal, just a warning to allow other "creative" solutions.
-# TODO: fix this with 'adduser' like in zephyr/docker-build.sh
-test "$(id -n)" = 1001 ||
-    >&2 printf "Warning: this script should be run as user ID 1001 to match the container\n"
+# The --user option below can cause the command to run as a user who
+# does not exist in the container. So far so good but in case something
+# ever goes wrong try replacing --user with the newer
+# scripts/sudo-cwd.sh script.
+test "$(id -u)" = 1000 ||
+  >&2 printf "Warning: this script should be run as user ID 1000 to match the container's account\n"
```
Comment (greg-intel) [inline scripts/docker-run.sh:33]: Out of curiosity, what changed the user from 1001 to 1000?

File: zephyr/docker-build.sh [hunk 3]
```diff
@@ -52,7 +52,7 @@ exec_as_sof_uid()
     # Double sudo to work around some funny restriction in
     # zephyr-build:/etc/sudoers: 'user' can do anything but... only as
     # root.
-    sudo sudo -u "$sof_user" "$0" "$@"
+    sudo sudo -u "$sof_user" http_proxy="$http_proxy"  https_proxy="$https_proxy" "$0" "$@"
```
Comment (greg-intel) [inline zephyr/docker-build.sh:55]: I know this all works and has been tested, but where does $http_proxy and $https_proxy come from here?

**Review by greg-intel — APPROVED** [review]

**Review by fredoh9 — APPROVED** [review]
looks good for stable-v2.2

File: .github/workflows/zephyr.yml (hunk 1 above)
Comment (marc-hb) [reply .github/workflows/zephyr.yml:31]: Yes because this is a stable branch, see commit message.

File: scripts/docker-run.sh (hunk 2 above)
Comment (marc-hb) [reply scripts/docker-run.sh:33]: This is a bug fix, I got the ID wrong. See commit message.

File: zephyr/docker-build.sh (hunk 3 above)
Comment (marc-hb) [reply zephyr/docker-build.sh:55]: This is a shell script so they come from your personal environment and (they're optional)

**Review by marc-hb — COMMENTED** [review]
Thanks for the reviews!

**Pull request merged**